### PR TITLE
Include path fragment in open api spec and routes

### LINKED ISSUE #779: Open API spec should include path fragment specified as --api-path  parameter
When starting Franklin with --api-path=mypath, the content of the Open api spec document doesn't reflect that the API is available as a deeper location then the web server root. This creates invalid links when you Try them out in a swagger frontend.

**Snippet of the current content of the open-api/spec.yaml (ignoring --api-path)**
```
openapi: 3.0.3
info:
  title: Franklin
  version: 0.0.1
paths:
  /collections:
    get:
      description: A list of collections
      operationId: collections
      responses:
        '200':
          description: ''
          content:
            application/json:
              schema:
                $ref: '#/components/schemas/Json'
  /collections/{p1}:
    get:
      description: A single collection
      operationId: collectionUnique
      parameters:
      - name: p1
        in: path
        required: true
        schema:
          type: string
      responses:
        '200':
          description: ''
          content:
            application/json:
              schema:
                $ref: '#/components/schemas/Json'
        '404':
          description: not found
          content:
            application/json:
              schema:
                $ref: '#/components/schemas/NotFound'
```
**Recommended content of the open-api/spec.yaml when --api-path is set**
```
openapi: 3.0.3
info:
  title: Franklin
  version: 0.0.1
paths:
  /mypath/collections:
    get:
      description: A list of collections
      operationId: collections
      responses:
        '200':
          description: ''
          content:
            application/json:
              schema:
                $ref: '#/components/schemas/Json'
  /mypath/collections/{p1}:
    get:
      description: A single collection
      operationId: collectionUnique
      parameters:
      - name: p1
        in: path
        required: true
        schema:
          type: string
      responses:
        '200':
          description: ''
          content:
            application/json:
              schema:
                $ref: '#/components/schemas/Json'
        '404':
          description: not found
          content:
            application/json:
              schema:
                $ref: '#/components/schemas/NotFound'
      ...
```

### PR DESCRIPTION
**Pull request by jisantuc**
## Overview

This PR makes it so that the API path fragment is respected everywhere. It accomplishes this through a helper method that takes an optional path prefix and the base path and builds a new endpoint input out of both.

### Checklist

- ~New tests have been added or existing tests have been modified~

### Testing Instructions

- Start the server with `/stac/api`, `stac/api`, and `stac/api/` `--api-path` arguments
- `http :9090/stac/api/collections` with each of them
- check `:9090/open-api/spec.yaml` for each of them
- start the server again with options like `--with-tiles` and `--with-transactions`
- check the spec again
- confirm that your path prefix is respected in all of the links

Closes #779  (if applicable)

### CODE REVIEW

File: application/src/main/scala/com/azavea/franklin/api/Server.scala [hunk 1]
```diff
@@ -134,10 +142,8 @@ $$$$
         ).routes
         landingPageRoutes = new LandingPageService[IO](apiConfig).routes
         router = CORS(
-          Router(
```
Comment (jisantuc) [inline application/src/main/scala/com/azavea/franklin/api/Server.scala:137]: `Router` can be used to map different services on different endpoints, but `tapir` is already doing that for us with the endpoint definitions / http4s interpreter, so we can unindent

File: application/src/main/scala/com/azavea/franklin/api/Server.scala [hunk 2]
```diff
@@ -98,21 +98,24 @@ $$$$
...
-        ].endpoints ++ new TileEndpoints[
+        ](apiConfig.path).endpoints ++ new TileEndpoints[
           IO
         ](
-          apiConfig.enableTiles
+          apiConfig.enableTiles,
+          apiConfig.path
         ).endpoints ++ landingPage.endpoints
-        docs      = allEndpoints.toOpenAPI("Franklin", "0.0.1")
+        docs      = OpenAPIDocsInterpreter.toOpenAPI(allEndpoints, "Franklin", "0.0.1")
```
Comment (jisantuc) [inline application/src/main/scala/com/azavea/franklin/api/Server.scala:118]: Get rid of a deprecation that's been annoying but unimportant for a while

**Comment by jisantuc** [comment]
Gonna go ahead and 🤠 this in since I'm out all of next week

**Review by pomadchin — COMMENTED** [review]
A clean and shiny PR 🚀 
🤦 pressing the button I forgot to press.

**Pull request merged**